amlb/datasets/openml.py

```diff
@@ -256,7 +256,15 @@ def _save_split(self, df, path, name):
 
     def _get_categorical_values(self, col):
         feat = next((f for f in self.ds._oml_dataset.features.values() if f.name == col), None)
-        return feat.nominal_values if feat is not None else None
+        if feat is not None:
+            # openml-python converts categorical features which look boolean to
+            # boolean values, which always write values as 'True' and 'False',
+            # so we need to adapt the header accordingly.
+            # Not doing so causes an issue in the R packages.
```

Comment:
Nice catch.
This autoconversion to boolean is new, isn't it? I remember that I quickly mentioned that a couple of months ago, and I don't think it was the case by then.
Btw, `H2OAutoML` is also using the arff splits and I didn't notice any issue there recently: any specific dataset that was failing? I just want to verify the behaviour in H2O, thanks.

Reply:
I don't know when exactly from the top of my head, it has indeed been changed between 0.10 (previous version we used) and 0.12.2. I had to fix this because the R packages failed on KC1 because of this. The header will say `{false, true}` but the data is `{True, False}`. Other frameworks seemed to ignore this difference.